Source/WebKit/UIProcess/Launcher/cocoa/ProcessLauncherCocoa.mm

```diff
@@ -56,8 +56,66 @@
 #import "CodeSigning.h"
 #endif
 
+#if USE(EXTENSIONKIT)
+#import "ExtensionKitSPI.h"
+
+SOFT_LINK_FRAMEWORK_OPTIONAL(ServiceExtensions);
+SOFT_LINK_CLASS_OPTIONAL(ServiceExtensions, _SEServiceConfiguration);
+SOFT_LINK_CLASS_OPTIONAL(ServiceExtensions, _SEServiceManager);
+#endif
+
 namespace WebKit {
 
+#if USE(EXTENSIONKIT)
+static std::pair<const char*, RetainPtr<NSString>> serviceNameAndIdentifier(ProcessLauncher::ProcessType processType)
+{
+    switch (processType) {
+    case ProcessLauncher::ProcessType::Web:
+        // FIXME: Add support for lockdown mode again.
+        return { "com.apple.WebKit.WebContent", @"com.apple.mobilesafari.WebContentExtension" };
+    case ProcessLauncher::ProcessType::Network:
+        return { "com.apple.WebKit.Networking", @"com.apple.mobilesafari.NetworkingExtension" };
+#if ENABLE(GPU_PROCESS)
+    case ProcessLauncher::ProcessType::GPU:
+        return { "com.apple.WebKit.GPU", @"com.apple.mobilesafari.GPUExtension" };
+#endif
+    }
+}
+
+static void launchWithExtensionKit(ProcessLauncher& processLauncher, ProcessLauncher::ProcessType processType, std::function<void(ThreadSafeWeakPtr<ProcessLauncher> weakProcessLauncher, _SEExtensionProcess* process, const char* name, NSError* error)> handler)
+{
+    auto [name, identifier] = serviceNameAndIdentifier(processType);
+
+    auto configuration = adoptNS([alloc_SEServiceConfigurationInstance() initWithServiceIdentifier:identifier.get()]);
+    _SEServiceManager* manager = [get_SEServiceManagerClass() performSelector:@selector(sharedInstance)];
+
+    switch (processType) {
+    case ProcessLauncher::ProcessType::Web: {
+        auto block = [handler = handler, weakProcessLauncher = ThreadSafeWeakPtr { processLauncher }, name = name](_SEContentProcess* _Nullable process, NSError* _Nullable error) {
+            handler(weakProcessLauncher, process, name, error);
+        };
+        [manager performSelector:@selector(contentProcessWithConfiguration:completion:) withObject:configuration.get() withObject:block];
```

Comment:
If `manager` does this work off the main thread, we might need to use isolated copy on `name`.

Reply:
Ditto for the other `case` sections.